stdlib/public/SDK/Foundation/Foundation.swift

```diff
@@ -95,3 +95,20 @@ extension CVarArg where Self: _ObjectiveCBridgeable {
     return _encodeBitsAsWords(object)
   }
 }
+
+//===----------------------------------------------------------------------===//
+// Runtime support for NSKeyedArchives
+//===----------------------------------------------------------------------===//
+
+@_silgen_name("swift_registerClassNameForArchiving")
+public func _registerClassNameForArchiving(_ NameForClass: UnsafePointer<CChar>,
+                                           _ Class: AnyClass) {
+  // If it's not possible to create a String from the name, it should abort
+  // and not fail silently.
+  let NameStr = String(validatingUTF8: NameForClass)!
+
+  // Register the class name mapping for archiving and unarchiving.
+  NSKeyedArchiver.setClassName(NameStr, for: Class)
+  NSKeyedUnarchiver.setClass(Class, forClassName: NameStr)
+}
```

Comment:
Nitpick, but this looks like the llvm naming conventions translated into Swift. This should have lowercase variable names.